Framework/Algorithms/src/Rebin.cpp

```diff
@@ -333,15 +331,15 @@ void Rebin::propagateMasks(API::MatrixWorkspace_const_sptr inputWS,
     masked_bins.push_back(XValues[(*it).first + 1]);
   }
 
-  // Create a zero vector for the errors because we don't care about them here
-  const MantidVec zeroes(weights.size(), 0.0);
-  // Create a vector to hold the redistributed weights
-  auto &XValues_new = outputWS->x(hist);
-  MantidVec newWeights(XValues_new.size() - 1), zeroes2(XValues_new.size() - 1);
+  //// Create a zero vector for the errors because we don't care about them here
+  auto errSize = weights.size();
+  Histogram oldHist(BinEdges(std::move(masked_bins)),
+                    Frequencies(std::move(weights)),
+                    FrequencyStandardDeviations(errSize, 0));
   // Use rebin function to redistribute the weights. Note that distribution flag
   // is set
-  VectorHelper::rebin(masked_bins, weights, zeroes, XValues_new.rawData(),
-                      newWeights, zeroes2, true);
+  auto newHist = HistogramData::rebin(oldHist, outputWS->binEdges(hist));
```

Comment:
Aren't there some more changes to do? Am I overlooking something or are you not using the result?